### PR TITLE
Update csv_to_sqlite.py to normalize dataset features, recreates and replaces ClassicHit.db, adds file normalized_ClassicHit.csv

### PR DESCRIPTION
**Pull request by riley-1995**

### CODE REVIEW

**Review by madisonewebb — APPROVED** [review]
LGTM!

File: assets/csv_to_sqlite.py [hunk 1]
```diff
@@ -1,20 +1,40 @@
...
+def normalize_features(df, save_to_csv=False):
+    features = ['Danceability', 'Energy', 'Loudness','Speechiness', 'Acousticness', 
+            'Instrumentalness', 'Liveness', 'Valence']
+    scaler = MinMaxScaler()
+    df2 = df.copy()
+    df2['Loudness_dB'] = df['Loudness'] # Keep the original Loudness column, which is in dB, but rename it
+    df2[features] = scaler.fit_transform(df[features])
 
+    if save_to_csv:
+        df2.to_csv(os.path.join(CSV_NORMALIZED_FILE), index=False)
```
Comment (Lomzem) [inline assets/csv_to_sqlite.py:27]: It might be cleaner to put this in another file

File: assets/csv_to_sqlite.py [hunk 2]
```diff
@@ -41,11 +62,12 @@ def create_database():
 
     c.execute(table_query)
```
Comment (Lomzem) [inline assets/csv_to_sqlite.py:63]: This is _really_ gross (talking about my own code). I know I started this initial database script like this but I was just rushing for our MVP. In the future (especially when we migrate the postgres) we should actually use an ORM like Python's `SQLAlchemy` for example

**Review by Lomzem — APPROVED** [review]
Good enough for a quickfix. Should definitely clean this up in future

**Review by avirdi1 — APPROVED** [review]
LGTM!

File: assets/csv_to_sqlite.py (hunk 1 above)
Comment (riley-1995) [reply assets/csv_to_sqlite.py]: Done! Can you check out the latest changes I made before merging it to main?

File: assets/CsvToSqlite.py [hunk 3]
```diff
@@ -36,25 +13,27 @@ def create_database(df):
...
+        CREATE TABLE music_data (
+        Song_ID INTEGER PRIMARY KEY,
         Title VARCHAR(255),
         Artist VARCHAR(255),
         Year INT, -- Year of release (assuming integer year)
         Duration INT, -- Duration of the track in milliseconds
         Time_Signature INT,
+        `Key` INT, -- Musical key as integer (e.g., 0 = C, 1 = C#, etc.)
+        Mode INT, -- Mode: 0 = Minor, 1 = Major
+        Key_String VARCHAR(10), -- Key and mode as string (e.g., "C Maj")
```
Comment (Lomzem) [inline assets/CsvToSqlite.py:25]: I disagree with this. It's unnecessary to store a string of 2^10 bytes when there's only 24 possibilities for key's. Also, it shouldn't be the database's job to turn the data into readable format, that should be the responsibility of our programs in ts or python. Your addition of Camelot_Key seems good enough.

**Review by Lomzem — CHANGES_REQUESTED** [review]

**Review by madisonewebb — APPROVED** [review]
LGTM!

**Comment by Lomzem** [comment]
@riley-1995 what do you think of this database design?

After some thinking, I actually supporting storing keyString in the database, but I think we should separate the key logic into a separate table.

```
Table Songs {
  songId integer [PK]
  title string
  artist string
  danceability float
  energy float
  loudness float
  camelotKey integer
}

Table CamelotKeys {
  camelotKey integer [PK]
  key integer
  mode integer
  keyString string
}

Ref: Songs.camelotKey > CamelotKeys.camelotKey
```
20250319_16h16m38s_grim

I was thinking because `key`, `mode`, and `keyString` all depend on camelotKey to determine what they will be, not `songId`. Then, `songId` determines what `camelotKey` will be

**Comment by riley-1995** [comment]
> @riley-1995 what do you think of this database design?
> 
> After some thinking, I actually support storing `keyString` in the database, but I think we should separate the key logic into a separate table.
> 
> ```
> Table Songs {
>   songId integer [PK]
>   title string
>   artist string
>   danceability float
>   energy float
>   loudness float
>   camelotKey integer
> }
> 
> Table CamelotKeys {
>   camelotKey integer [PK]
>   key integer
>   mode integer
>   keyString string
> }
> 
> Ref: Songs.camelotKey > CamelotKeys.camelotKey
> ```
> 
> 20250319_16h16m38s_grim
> 
> I was thinking because `key`, `mode`, and `keyString` all depend on camelotKey to determine what they will be, not `songId`. Then, `songId` determines what `camelotKey` will be

I like it! I have updated the files having to do with the db accordingly.
